### PR TITLE
Remove deprecated storage.tsdb.retention flag

### PR DESCRIPTION
**Pull request by roidelapluie**

### CODE REVIEW

**Review by beorn7 — APPROVED** [review]

File: cmd/prometheus/main.go [hunk 1]
```diff
@@ -362,11 +357,9 @@ func main() {
 		"Size at which to split the tsdb WAL segment files. Example: 100MB").
 		Hidden().PlaceHolder("<bytes>").BytesVar(&cfg.tsdb.WALSegmentSize)
 
-	serverOnlyFlag(a, "storage.tsdb.retention", "[DEPRECATED] How long to retain samples in storage. This flag has been deprecated, use \"storage.tsdb.retention.time\" instead.").
-		SetValue(&oldFlagRetentionDuration)
-
+	var flagRetentionDuration model.Duration
 	serverOnlyFlag(a, "storage.tsdb.retention.time", "How long to retain samples in storage. When this flag is set it overrides \"storage.tsdb.retention\". If neither this flag nor \"storage.tsdb.retention\" nor \"storage.tsdb.retention.size\" is set, the retention time defaults to "+defaultRetentionString+". Units Supported: y, w, d, h, m, s, ms.").
```
Comment (machine424) [inline cmd/prometheus/main.go:361]:
```suggestion
	serverOnlyFlag(a, "storage.tsdb.retention.time", "How long to retain samples in storage. If neither this flag nor \"storage.tsdb.retention.size\" is set, the retention time defaults to "+defaultRetentionString+". Units Supported: y, w, d, h, m, s, ms.").
```

File: cmd/prometheus/main.go [hunk 2]
```diff
@@ -362,11 +357,9 @@ func main() {
...
-		SetValue(&newFlagRetentionDuration)
+		SetValue(&flagRetentionDuration)
```
Comment (machine424) [inline cmd/prometheus/main.go:362]: I think we can directly
```suggestion
		SetValue(&cfg.tsdb.RetentionDuration)
```

**Review by machine424 — COMMENTED** [review]
Thanks!

**Review by machine424 — APPROVED** [review]
lgtm, thanks!
(I can still see a `nor nor`, but that could be the reason for someone to open their first PR ;))

**Review by machine424 — APPROVED** [review]

**Pull request merged**